cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/transform/LogParserTransform.java

```diff
@@ -115,13 +118,35 @@ public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter)
         return;
       }
       output = parseRequest(logMatcher, S3_INDICES);
-    } else {
+    } else if (CLF_LOG.equals(config.logFormat)) {
       Matcher logMatcher = CLF_LOG_PATTERN.matcher(log);
       if (!logMatcher.matches() || logMatcher.groupCount() < CLF_REGEX_LENGTH) {
         LOG.debug("Couldn't parse log because the log did not match the CLF format. log: {}", log);
         return;
       }
       output = parseRequest(logMatcher, CLF_INDICES);
+    } else {
+      if (log.startsWith("#")) {
+        LOG.trace("Log is a comment. Ignoring...");
+        return;
+      }
+
+      String[] fields = log.split("\\t");
+      String uri = fields[7];
+      String ip = fields[4];
+      long ts = sdfCloudfront.parse(String.format("%s:%s UTC", fields[0], fields[1])).getTime();
+      UserAgentStringParser parser = UADetectorServiceFactory.getResourceModuleParser();
+      ReadableUserAgent userAgent = parser.parse(fields[10]);
+      String browser = userAgent.getFamily().getName();
```

Comment:
can ``userAgent.getFamily()`` or ``userAgent.getDeviceCategory()`` or ``userAgent.getDeviceCategory().getCategory()`` be null?